sashimi/gui/status_bar.py

```diff
@@ -0,0 +1,104 @@
+from PyQt5.QtWidgets import (
+    QLabel,
+    QStatusBar,
+    QProgressBar
+)
+
+from sashimi.state import GlobalState, State, get_voxel_size
+from sashimi.config import read_config
+
+conf = read_config()
+
+
+class StatusBarWidget(QStatusBar):
+    def __init__(self, state: State, timer):
+        super().__init__()
+        self.state = state
+        self.timer = timer
+
+        self.framerate_lbl = QLabel("0 Hz")
+        self.frame_size_lbl = QLabel()
+        self.voxel_size_lbl = QLabel()
+        self.interplane_lbl = QLabel()
+        self.experiment_progress = QProgressBar()
+        self.experiment_progress.setFormat("Volume %v of %m")
+        self.lbl_experiment_progress = QLabel()
+        self.experiment_progress.hide()
+        self.lbl_experiment_progress.hide()
+
+        self.addPermanentWidget(self.framerate_lbl)
+        self.addPermanentWidget(self.frame_size_lbl)
+        self.addPermanentWidget(self.voxel_size_lbl)
+        self.addPermanentWidget(self.interplane_lbl)
+        self.addPermanentWidget(self.experiment_progress)
+        self.addPermanentWidget(self.lbl_experiment_progress)
+
+        self.voxel_size = None
+
+        self.timer.timeout.connect(self.update_all_labels)
+
+    def update_all_labels(self):
+        self.update_framerate_view()
+        self.update_frame_size()
+        self.update_voxel_size()
+        self.refresh_progress_bar()
+        if self.state.global_state == GlobalState.PAUSED:
+            self.hide()
+        else:
+            self.show()
+
+    def update_framerate_view(self):
+        """Update the framerate and check whether it is fast enough for the current
+        parameters configuration.
+        """
+
+        frame_rate = self.state.get_triggered_frame_rate()
+        if frame_rate is not None:
+            self.framerate_lbl.setText(
+                "Framerate: {} Hz".format(round(frame_rate, 2))
+            )
+
+            # Find the expected framerate depending on the global state
+            if self.state.global_state == GlobalState.PREVIEW:
+                expected_frame_rate = None
+            if self.state.global_state == GlobalState.VOLUME_PREVIEW:
+                vol_sets = self.state.volume_setting
+                n_planes = (vol_sets.n_planes - vol_sets.n_skip_start - vol_sets.n_skip_end)
+                expected_frame_rate = vol_sets.frequency * n_planes
+            if self.state.global_state == GlobalState.PLANAR_PREVIEW:
+                expected_frame_rate = self.state.single_plane_settings.frequency
```

Comment:
this calculation doesn't seem like something that should be in the GUI code (rather state)

Reply:
but the frame rate is calculated for displaying purposes only, that is the reason this was here I guess. You still stand behind the point?